### DISCUSSION TITLE
Next is picking the wrong dynamic route and seemingly ignoring getStaticPaths

### DISCUSSION THREAD

**Discussion by tremby** [vercel/next.js#22448] [Help]
I have two dynamic routes:

1. `pages/[...uids].tsx`, where `uids` can be one or more parts long.
2. `pages/blog/[...segments].tsx`, where `segments` is usually four parts long but I want this route to still catch if the pattern matches with a single part and the previous route does not match. (This is useful in preview mode, when the blog post doesn't yet have year/month/date info, which would usually be the first three parts of `segments`.)

Both have `getStaticPaths` carefully set, returning the list of all pages for each route.

One of the `pages/[...uids].tsx` paths is `/blog/news/`, which `getStaticPaths` of the first route correctly includes. The `getStaticPaths` response of the second route does not contain this path.

During client-side navigation to `/blog/news/` I'm seeing it incorrectly choose the second route, and so render the page incorrectly.

If I hit refresh on that page it correctly chooses the first route.

Is this a bug or am I doing something wrong? What's the point of `getStaticPaths` if it isn't used to choose the route?

**Comment by nilsborg** [comment]
We're having exactly the same issue!
The only thing that got this to work also on client side navigation was to specifically exclude the slug in the catch-all route and create that page as new file with that slug as the filename...
